client/pool.c

```diff
@@ -1419,3 +1427,27 @@ void* pool_remove_inactive_connections(void* arg)
 	return NULL;
 }
 
+static inline void update_mean_log_diff(struct connection_pool_data *conn_data, struct xdag_pool_task *task, xdag_hash_t hash)
+{
+	const xdag_time_t task_time = task->task_time;
+
+	if(conn_data->task_time < task_time) {
+		moving_average(&conn_data->mean_log_difficulty, diff2log(hash2difficulty(conn_data->last_min_hash)),conn_data->bounded_task_counter); 
+                if(conn_data->bounded_task_counter<NSAMPLES_MAX)
+			++conn_data->bounded_task_counter;
+		memcpy(conn_data->last_min_hash,hash,sizeof(xdag_hash_t));
+	} else if(xdag_cmphash(hash, conn_data->last_min_hash) < 0)
+		memcpy(conn_data->last_min_hash,hash,sizeof(xdag_hash_t));
+
+	if(conn_data->miner) {
```

Comment:
conn_data->miner cannot be NULL here.

Reply:
You did it this way in your function too

Reply:
you do it in calculate_nopaid_shares